pandas/core/indexes/multi.py

```diff
@@ -2890,6 +2890,24 @@ def convert_indexer(start, stop, step, indexer=indexer, codes=level_codes):
 
             return m
 
+        def search_code_location(code):
+            # Base on argument ``code``, search where is ``code`` in level's codes
+
+            if level > 0 or self.lexsort_depth == 0:
+                # Desired level is not sorted
+                locs = np.array(level_codes == code, dtype=bool, copy=False)
+                if not locs.any():
+                    # The label is present in self.levels[level] but unused:
+                    raise KeyError(key)
+                return locs
+
+            i = level_codes.searchsorted(code, side="left")
+            j = level_codes.searchsorted(code, side="right")
+            if i == j:
+                # The label is present in self.levels[level] but unused:
+                raise KeyError(key)
+            return slice(i, j)
+
```

Comment:
Do you really need this nested method? Unless I'm mistaken, you can just put the ``code = level_index.get_loc(key)`` in a conditional statement.

Reply:
there is  `code = level_index.get_loc(key)` after else statement. This code existed before. I just nest code and put it upper place. Because nesting it shows more clearly what code is for.

Reply:
>  there is `code = level_index.get_loc(key)` after else statement. This code existed before.

Yes, that's what I was referring to: this PR could have just replaced that with
```python3
if not is_list_like(key) and isna(key):
    code = -1
else:
    code = level_index.get_loc(key)
```
right?


> Because nesting it shows more clearly what code is for.

In my (limited) experience, nested functions make code _less_ readable, and harder to debug. If we want to isolate this code, I would prefer with a separate (better documented) function. But maybe it's a matter of taste, @jreback what's your take on this?